05.wc/wc.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options, file_paths = parse_options(ARGV)
+
+  # パイプで接続した場合 $stdin.tty? は false になる
+  if file_paths.empty? && !$stdin.tty?
+    file_stats = count_content($stdin.read)
+    file_stats[:file_path] = nil
+    file_stats_list = [file_stats]
+    total_stats = nil
+  else
+    file_stats_list, total_stats = collect_file_stats(file_paths)
+  end
```

Comment:
```suggestion
  file_stats_list, total_stats = file_paths.empty? && !$stdin.tty? ? hogehoge : collect_file_stats(file_paths)
```

標準入力の情報を使う場合もメソッド化してあげるとか

Reply:
メソッド化しました。
https://github.com/cellotak/ruby-practices/pull/12/commits/1ca7719ef95f06d9427583d179ef31c7c69dadf0